lib/dues_calculator.rb

```diff
@@ -9,16 +9,22 @@ def self.dues_per_competitor(country_iso2, base_entry_fee_lowest_denomination, c
   end
 
   def self.dues_per_competitor_in_usd(country_iso2, base_entry_fee_lowest_denomination, currency_code)
-    country_band = CountryBand.find_by(iso2: country_iso2)&.number
+    country_band = CountryBand.find_by(iso2: country_iso2)
+    country_band_detail = country_band&.active_country_band_detail
+    registration_fees_in_usd = Money.new(base_entry_fee_lowest_denomination, currency_code).exchange_to("USD")
 
-    input_money_us_dollars = Money.new(base_entry_fee_lowest_denomination, currency_code).exchange_to("USD")
+    return nil if country_band_detail.nil?
 
-    registration_fee_dues_us_dollars = input_money_us_dollars * CountryBand.percent_registration_fee_used_for_due_amount(country_band)
-    country_band_dues_us_dollars = country_band.present? && country_band > 0 ? CountryBand::BANDS[country_band][:value] : 0
-    # times 100 because Money require lowest currency subunit, which is cents for USD
-    country_band_dues_us_dollars_money = Money.new(country_band_dues_us_dollars * 100, "USD")
+    # Calculation of 'registration fee dues'
+    due_percent_registration_fee = country_band_detail.due_percent_registration_fee
```

Comment:
Is declaring this variable super-duper necessary? It's only used once below...

Reply:
Okay removed.